src/rtctools/data/timeseries.py

```diff
@@ -0,0 +1,88 @@
+"""Module for processing timeseries data."""
```

Comment:
There is a Timeseries class in `rtctools.optimization`, but the new functionalities in this PR are not specific for optimization but could also be used for simulation. Therefore, I figured it makes more sense to put them in `rtctools.data`. Perhaps it would also make sense to move the Timeseries class to `rtctools.data`, but that would be a breaking change.

Reply:
If 'get_timeseries_from_{csv,pi}' move back to the format-specific modules, the remaining module could be renamed to 'util.py'. This would be less confusing than having multiple modules named 'timeseries' within the same package.

Reply:
Sounds reasonable. I'll rename this file and move the read functions to their respective Mixin module.